src/hotspot/share/opto/vectornode.cpp

```diff
@@ -854,6 +855,10 @@ bool VectorNode::is_all_ones_vector(Node* n) {
   case Op_ReplicateL:
   case Op_MaskAll:
     return is_con(n->in(1), -1);
+  case Op_ReplicateF:
+    return n->in(1)->bottom_type() == TypeF::ONE;
+  case Op_ReplicateD:
+    return n->in(1)->bottom_type() == TypeD::ONE;
```

Comment:
`is_all_ones_vector` here means all bits are `1` (i.e. `-1` for integer types). So `TypeF::ONE` is not the right value here?

Reply:
Missed this comment?